presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -194,4 +218,22 @@ public Symbol getRight()
             return right;
         }
     }
+
+    public boolean isDistributed()
+    {
+        return method == DISTRIBUTED;
+    }
+
+    public static boolean canDistributeJoin(Session session, PlanNode rightNode, List<JoinNode.EquiJoinClause> equiJoinClauses, JoinNode.Type type)
```

Comment:
AFAIK code never does broadcast joins if `distributed_joins` is set.
This property should probably be deprecated and replaced with something like `join_method=BROADCAST|DISTRIBUTED|AUTOMATIC`.